content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1708 @@
+
+- It returns the values with the keys `student_id`, `first_name`, `last_name`, `birth_date`, and `gender` from `$row`.
+
+- It displays the values with the keys `student_id`, `first_name`, `last_name`, `birth_date`, and `gender` from `$row`.
+
+- It accesses the values `student_id`, `first_name`, `last_name`, `birth_date`, and `gender` from the `students` table.
+
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php], init=[commands/PreparedStatement/UpdateFemaleGender.sql]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $data = array(
+        'gender' => 'F',
+        'female_gender' => 'Female'
+    );
+    
+    try {
+        $sql = "UPDATE students SET gender=:gender WHERE gender=:female_gender";
+        $pstmt = $conn->prepare($sql);
+        if (!$pstmt->execute($data)) {
+            throw new Exception("Unable to update values in the table.");
+        }
+        echo "Successfully updated values in the table.";
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `Unable to update values in the table.`.
+
+- It prints `Successfully connected to the database.` and `Successfully updated values in the table.`.
+
+
+/// type=SS, answer=[4]
+
+Which statement correctly describes the code on lines 4, 5, 6 and 7 of `student.php`?
+
+- It adds the associative array named `$data` to the database.
+
+- It holds one value in each key of `gender` and `female_gender`.
+
+- It accesses the elements of `$data` using the `array()` construct.
+
+- It creates the associative array named `$data` with two keys `gender` and `female_gender`.
+
+- It creates the associative array named `$data` with two elements `gender` and `female_gender`.
+
+
+/// type=MS, answer=[3,4]
+
+Which statements correctly describe `UPDATE students SET gender=:gender WHERE gender=:female_gender` on line 10 of `student.php`?
+
+- It modifies all rows in the `gender` column.
+
+- It selects the `students` table to be deleted.
+
+- It selects the `students` table to be modified.
+
+- It updates the rows in the `gender` column with the values set by the named placeholders `:gender` and `:female_gender`.
```

Comment:
Please improve this statement... Please check this out:

> It updates the rows in the `gender` column with the value of the named placeholder `:gender`.
> It sets the condition that all rows containing values equal to the value of the named placeholder `:female_gender` are updated.

These are the possible answer that correctly describe the SQL statement above. Please check. Thanks.